tests/Aspire.Dashboard.Tests/Telemetry/ComponentTelemetryContextTests.cs

```diff
@@ -0,0 +1,86 @@
+        telemetryContext.UpdateTelemetryProperties([new ComponentTelemetryProperty("Test", new AspireTelemetryProperty("Value"))]);
+        Assert.Equal(3, telemetryContext.Properties.Count);
+        Assert.True(telemetrySender.ContextChannel.Reader.TryRead(out parametersUpdateOperation));
+        Assert.Equal("/telemetry/operation - $aspire/dashboard/component/paramsSet", parametersUpdateOperation.Name);
+        Assert.Single(parametersUpdateOperation.Properties);
+
+        // If value didn't change, we shouldn't post again
+        telemetryContext.UpdateTelemetryProperties([new ComponentTelemetryProperty("Test", new AspireTelemetryProperty("Value"))]);
+        Assert.Equal(3, telemetryContext.Properties.Count);
+        Assert.False(telemetrySender.ContextChannel.Reader.TryRead(out parametersUpdateOperation));
+
+        telemetryContext.UpdateTelemetryProperties([new ComponentTelemetryProperty("Test", new AspireTelemetryProperty("NewValue"))]);
+        Assert.Equal(3, telemetryContext.Properties.Count);
+        Assert.True(telemetrySender.ContextChannel.Reader.TryRead(out parametersUpdateOperation));
+
+        telemetryContext.Dispose();
+        Assert.True(telemetrySender.ContextChannel.Reader.TryRead(out var disposeOperation));
+        Assert.Equal("/telemetry/operation - $aspire/dashboard/component/dispose", disposeOperation.Name);
+    }
+
+    [Fact]
+    public async Task ComponentTelemetryContext_TelemetryDisabled_EndToEnd()
+    {
+        var telemetryContext = new ComponentTelemetryContext(nameof(ComponentTelemetryContextTests));
+        var telemetrySender = new TestDashboardTelemetrySender { IsTelemetryEnabled = false };
+        var telemetryService = new DashboardTelemetryService(NullLogger<DashboardTelemetryService>.Instance, telemetrySender);
+
+        await telemetryContext.InitializeAsync(telemetryService, new TestJSRuntime());
+        Assert.False(telemetrySender.ContextChannel.Reader.TryRead(out _));
+
+        telemetryContext.UpdateTelemetryProperties([new ComponentTelemetryProperty("Test", new AspireTelemetryProperty("Value"))]);
+        Assert.Equal(3, telemetryContext.Properties.Count);
+        Assert.False(telemetrySender.ContextChannel.Reader.TryRead(out _));
+
+        telemetryContext.Dispose();
+        Assert.False(telemetrySender.ContextChannel.Reader.TryRead(out _));
+    }
+
+    private sealed class TestJSRuntime : IJSRuntime
+    {
+        public ValueTask<TValue> InvokeAsync<TValue>(string identifier, object?[]? args)
+        {
+            return typeof(TValue) == typeof(string) ? new ValueTask<TValue>((TValue)(object)string.Empty) : new ValueTask<TValue>(default(TValue)!);
+        }
+
+        public ValueTask<TValue> InvokeAsync<TValue>(string identifier, CancellationToken cancellationToken, object?[]? args)
+        {
+            return typeof(TValue) == typeof(string) ? new ValueTask<TValue>((TValue)(object)string.Empty) : new ValueTask<TValue>(default(TValue)!);
+        }
+    }
```

Comment:
This is browser/components logic leaking into models. They should be kept separated.